src/listeners.ts

```diff
@@ -2,12 +2,43 @@ import { Context } from "probot";
 import Webhooks from "@octokit/webhooks";
 import { lintCommitMessage } from "./lint";
 import { Commit as GithubCommit } from "types/github";
+import { findOrCreateUser } from "./models/User";
+import { createCommit, findCommit } from "./models/Commit";
+import { createCommitDiagnosis } from "./models/CommitDiagnosis";
 
-async function processCommit(commit: GithubCommit) {
-  const message = commit.message;
-  const lintOutput = await lintCommitMessage(message);
-  // TODO: save lint results and commit info in db
-  console.log(lintOutput.violations);
+async function processCommit(githubCommit: GithubCommit): Promise<void> {
+  const commitExists = await findCommit(githubCommit.id);
+  if (!!commitExists) {
+    return;
+  }
+  const message = githubCommit.message;
+  const { score, violations } = await lintCommitMessage(message);
+  const userName = githubCommit.author.name;
+  const user = await findOrCreateUser(userName);
```

Comment:
Git commit username & email is a customizable name of the committer, and may differ in different commits. A user may have multiple emails too (e.g. a real email & no-reply email generated by GitHub). To be precise, we should be resolving the GitHub user to associate with the correct user.

Reply:
Do you know what the different between the `name` field and the `username` field is? Github docs don't display the `username` field but the webhook includes it in the payload.

Reply:
I *guess* it's the GitHub user username, but that can change too. Can we resolve the unique ID of the GitHub user if possible?

Reply:
it's not in the payload. Do you want to call the Github User API?

Reply:
I don't mind, you can decide whether to call the APIs. Changing username is rare so we may ignore it for now, but we should at least prepare for this case in future.